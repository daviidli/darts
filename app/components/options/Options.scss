.options {
	.button {
		width: 4em;
		height: 3em;
		background-color: brown;
		font-size: 3em;
	}
}
