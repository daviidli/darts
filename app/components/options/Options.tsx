import React from 'react';
import { Grid, Button } from '@material-ui/core';
import styles from './Options.scss';

export type optionsItems = {
	label: string;
	onClick: (label: string) => any;
};

type Props = {
	items: optionsItems[];
};

const Options = (props: Props) => {
	const { items } = props;
	return (
		<Grid container className={styles.options}>
			{items.map(item => (
				<Grid item key={item.label}>
					<Button
						className={styles.button}
						onClick={() => item.onClick(item.label)}
					>
						{item.label}
					</Button>
				</Grid>
			))}
		</Grid>
	);
};

export default Options;
