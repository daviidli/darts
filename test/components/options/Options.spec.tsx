import { spy } from 'sinon';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Options, { optionsItems } from '../../../app/components/options/Options';

Enzyme.configure({ adapter: new Adapter() });

const setup = (items: optionsItems[] = []) => {
	const wrapper = shallow(<Options items={items} />);
	return {
		wrapper,
		button: wrapper.find('.button')
	};
};

describe('Options component', () => {
	it('should match snapshot', () => {
		const items: optionsItems[] = [
			{
				label: 'op1',
				onClick: () => {}
			},
			{
				label: 'op2',
				onClick: () => {}
			}
		];
		const tree = renderer.create(<Options items={items} />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should have 1 button', () => {
		const onClick = spy();
		const items: optionsItems[] = [
			{
				label: 'firstButton',
				onClick
			}
		];

		const { button } = setup(items);
		expect(button.at(0).text()).toMatch('firstButton');
		button.at(0).simulate('click');
		expect(onClick.calledOnce).toBeTruthy();
	});

	it('should have 2 buttons', () => {
		const onClick1 = spy();
		const onClick2 = spy();

		const items: optionsItems[] = [
			{
				label: 'firstButton',
				onClick: onClick1
			},
			{
				label: 'secondButton',
				onClick: onClick2
			}
		];

		const { button } = setup(items);
		expect(button.at(0).text()).toMatch('firstButton');
		expect(button.at(1).text()).toMatch('secondButton');
		button.at(1).simulate('click');
		expect(onClick1.notCalled).toBeTruthy();
		expect(onClick2.calledOnce).toBeTruthy();
	});
});
