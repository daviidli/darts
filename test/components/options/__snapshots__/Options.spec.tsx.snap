// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options component should match snapshot 1`] = `
<WithStyles(ForwardRef(Grid))
  className="options"
  container={true}
  justify="center"
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    key="op1"
  >
    <WithStyles(ForwardRef(Button))
      className="button"
      color="primary"
      onClick={[Function]}
      variant="contained"
    >
      op1
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    key="op2"
  >
    <WithStyles(ForwardRef(Button))
      className="button"
      color="primary"
      onClick={[Function]}
      variant="contained"
    >
      op2
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
